packages/components/src/test/tree/controlled/UseUnifiedSelection.test.ts

```diff
@@ -399,43 +400,123 @@ describe("UnifiedSelectionEventHandler", () => {
 
+        // verify nodes selected based on updated unified selection
         expect(modelSource.getModel().getNode(nodes[0].id)?.isSelected).to.be.true;
         expect(modelSource.getModel().getNode(nodes[1].id)?.isSelected).to.be.true;
         expect(modelSource.getModel().getNode(nodes[2].id)?.isSelected).to.be.true;
       });
     });
 
     it("deselects nodes according unified selection", () => {
-      const nodes: TreeModelNodeInput[] = [createNode(), createNode(createTestECClassGroupingNodeKey)];
+      const nodes: TreeModelNodeInput[] = [
+        createNode(() => createTestECInstancesNodeKey({ instanceKeys: [{ id: "0x1", className: "Schema:Class" }] }), { id: "node_1" }),
+        createNode(createTestECClassGroupingNodeKey, { id: "node_2" }),
+      ];
 
       modelSource.modifyModel((model) => {
         model.setChildren(undefined, nodes, 0);
-        model.getNode(nodes[0].id)!.isSelected = true;
       });
 
-      selectionManager.getSelection.returns(new KeySet());
+      // setup initials selection
+      selectionManager.getSelection.returns(new KeySet(SelectionHelper.getKeysForSelection([getItemKey(nodes[0].item)])));
 
       using(createHandler(), (_) => {
+        // verify nodes selected based on initial unified selection
+        expect(modelSource.getModel().getNode(nodes[0].id)?.isSelected).to.be.true;
+        expect(modelSource.getModel().getNode(nodes[1].id)?.isSelected).to.be.false;
+
+        selectionManager.getSelection.reset();
+        selectionManager.getSelection.returns(new KeySet());
         selectionChangeEvent.raiseEvent(selectionEvent, selectionProvider);
 
+        // verify nodes selected based on updated unified selection
+        expect(modelSource.getModel().getNode(nodes[0].id)?.isSelected).to.be.false;
+        expect(modelSource.getModel().getNode(nodes[1].id)?.isSelected).to.be.false;
+      });
+    });
+
+    it("ignores selection changes on different imodel", () => {
+      const nodes: TreeModelNodeInput[] = [
+        createNode(() => createTestECInstancesNodeKey({ instanceKeys: [{ id: "0x1", className: "Schema:Class" }] }), { id: "node_1" }),
+        createNode(createTestECClassGroupingNodeKey, { id: "node_2" }),
+      ];
+
+      modelSource.modifyModel((model) => {
+        model.setChildren(undefined, nodes, 0);
+      });
+
+      // setup initials selection
+      selectionManager.getSelection.returns(new KeySet());
+
+      using(createHandler(), (_) => {
+        // verify nodes selected based on initial unified selection
+        expect(modelSource.getModel().getNode(nodes[0].id)?.isSelected).to.be.false;
+        expect(modelSource.getModel().getNode(nodes[1].id)?.isSelected).to.be.false;
+
+        const selectionKeys = SelectionHelper.getKeysForSelection(nodes.map((n) => getItemKey(n.item)));
+        selectionManager.getSelection.reset();
+        selectionManager.getSelection.returns(new KeySet(selectionKeys));
```

Comment:
not sure `getSelection` needs to return a different result here - if the selection changes for a different iModel, then `getSelection` is going to return its old result